examples/keccak.rs

```diff
@@ -2210,25 +2211,28 @@ struct CircuitParams {
 fn keccak_super_circuit<F: PrimeField<Repr = [u8; 32]> + Eq + Hash>(
     input_len: usize,
 ) -> SuperCircuit<F, KeccakCircuit> {
-    super_circuit::<F, KeccakCircuit, _>("keccak", |ctx| {
+    super_circuit::<F, KeccakCircuit, _, WitnessTraceGenerator<F>>("keccak", |ctx| {
         let in_n = (input_len * 8 + 1 + RATE_IN_BITS as usize) / RATE_IN_BITS as usize;
         let step_num = in_n * (1 + NUM_ROUNDS as usize);
 
         let single_config = config(SingleRowCellManager {}, SimpleStepSelectorBuilder {});
         // config(SingleRowCellManager {}, LogNSelectorBuilder {});
 
-        let (_, constants_table) = ctx.sub_circuit(
+        let (_, constants_table): (AssignmentGenerator<F>, LookupTable) = ctx.sub_circuit(
```

Comment:
why do we need to make the type explicit now?

Reply:
Because sub_circuit is generic by AssigmentGenerator, so we either need to put generics in sub_circuit or in the return type (return is shorter so I chose that).